subsys/bluetooth/shell/has.c

```diff
@@ -0,0 +1,112 @@
+/**
+ * @file
+ * @brief Bluetooth Hearing Access Service (HAS) shell.
+ *
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/audio/has.h>
+#include <shell/shell.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "bt.h"
+
+static int cmd_preset_reg(const struct shell *sh, size_t argc, char **argv)
+{
+	int err = 0;
+	struct bt_has_preset_register_param param = {
+		.index = shell_strtoul(argv[1], 16, &err),
+		.properties = shell_strtoul(argv[2], 16, &err),
+		.name = argv[3],
+	};
+
+	if (err < 0) {
+		shell_print(sh, "Invalid command parameter (err %d)", err);
+		return err;
+	}
+
+	err = bt_has_preset_register(&param);
+	if (err < 0) {
+		shell_error(sh, "Preset register failed (err %d)", err);
+		return err;
+	}
+
+	return 0;
+}
+
+static int cmd_preset_unreg(const struct shell *sh, size_t argc, char **argv)
+{
+	int err = 0;
+	const uint8_t index = shell_strtoul(argv[1], 16, &err);
```

Comment:
`shell_strtoul` returns a `unsigned long`, so while it's nice that we do check error now, we don't handle overflow when assigning the result to a `uint8_t`. 

For me it's OK as is, but that's something that can be approved upon (we could even consider adding functions such as `shell_strtou8`/`shell_strotuint8` to handle such cases.

Reply:
Yes, I thought about that, but I would not worry since this is test application.

Reply:
Agreed. It's something that can be added later, but my point is that checking a valid value range for overflow is just as important as checking the new error result IMO :)